File: pojo/src/main/java/com/escapsule/thalitera/entity/MeetingRoom.java
package com.escapsule.thalitera.entity;

import com.baomidou.mybatisplus.annotation.TableField;
import com.escapsule.thalitera.handler.PgUUIDTypeHandler;
import com.escapsule.thalitera.json.MeetingRoomFacilities;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.io.Serializable;
import java.time.OffsetDateTime;
import java.util.UUID;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class MeetingRoom implements Serializable {

    private String image;

    @TableField(value = "room_id", typeHandler = PgUUIDTypeHandler.class)
    private UUID roomId;

    /**
     * Readable name of the meeting room
     */
    private String name;

    private int capacityMin;

    private int capacityMax;

    private String building;

    private int floor;

    /**
     * See {@link com.escapsule.thalitera.constant.MeetingRoomStatusConstant}
     */
    private String status;

    /**
     * See {@link com.escapsule.thalitera.json.MeetingRoomFacilities}
     */
    private MeetingRoomFacilities facilities;

    /**
     * Created by user ID
     */
    private UUID createdBy;

    private OffsetDateTime createdAt;

    private OffsetDateTime updatedAt;

}

File: src/types/meeting-room.ts
export interface MeetingRoomFacilities {
  projecter?: boolean;
  whiteboard?: number;
  power_sockets?: number;
  coffee_break?: boolean;
  special_notes?: string[];
}

export interface MeetingRoom {
  room_id: string;
  image: string;
  name: string;
  capacity_min: number;
  capacity_max: number;
  building: string;
  floor: string;
  facilities: MeetingRoomFacilities;
}

export interface MeetingRoomResponse {
  code: number;
  message: string;
  data: MeetingRoom[];
  timestamp: number;
} 